core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala

```diff
@@ -405,7 +404,7 @@ class JobProgressListener(conf: SparkConf) extends SparkListener with Logging {
         updateAggregateMetrics(stageData, info.executorId, m, oldMetrics)
       }
 
-      val taskData = stageData.taskData.getOrElseUpdate(info.taskId, TaskUIData(info, None))
```

Comment:
Important note here: in the old code, the `elseUpdate` branch would only be taken in rare error cases where we  somehow purged the TaskUIData which should have been created when the task launched. It technically doesn't matter what we put in for the `Option[Metrics]` here since it just gets unconditionally overwritten on line 410 in the old code. So while my new code constructs TaskUIData with default metrics it doesn't actually change the behavior of this block.